SingularityBase/src/main/java/com/hubspot/mesos/MesosUtils.java

```diff
@@ -39,17 +42,44 @@ private static double getScalar(Resource r) {
     return r.getScalar().getValue();
   }
 
-  private static double getScalar(List<Resource> resources, String name) {
+  private static double getScalar(List<Resource> resources, String name, Optional<String> requiredRole) {
     for (Resource r : resources) {
 
-      if (r.hasName() && r.getName().equals(name) && r.hasScalar()) {
+      if (r.hasName() && r.getName().equals(name) && r.hasScalar() && isRequiredRole(r, requiredRole)) {
         return getScalar(r);
       }
     }
 
     return 0;
   }
 
+  private static Boolean hasRole(Resource r) {
```

Comment:
can you switch to primitive `boolean`s instead? removes the potential for NullPointerExceptions